### PR TITLE
Removed mut from the UdpSocket API.

### PR DESCRIPTION
**Pull request by TimonPost**
#115

### CODE REVIEW

File: src/net/udp.rs [hunk 1]
```diff
@@ -47,11 +45,12 @@ impl UdpSocket {
     }
 
     /// Receives a single datagram message on the socket. On success, returns the packet containing origin and data.
-    pub fn recv(&mut self) -> NetworkResult<Option<Packet>> {
-        let (len, addr) = self.socket.recv_from(&mut self.recv_buffer)?;
+    pub fn recv(&self) -> NetworkResult<Option<Packet>> {
+        let mut recv_buffer = vec![0; self._config.receive_buffer_max_size];
```
Comment (LucioFranco) [inline src/net/udp.rs]: so this means everytime we call `recv` we allocate a new buffer? Doesn't sound very efficient.

Comment (TimonPost) [reply src/net/udp.rs]: True if you have alternatives please let me know under the linked issue.

Comment (TimonPost) [reply src/net/udp.rs]: I'm not sure on how to solve this problem right way other then doing this

**Comment by LucioFranco** [comment]
I am going to close this as this is not the correct way to go about this.

**Pull request closed without merging**